public/app/features/search/state/SearchStateManager.test.ts

```diff
@@ -65,10 +65,12 @@ describe('SearchStateManager', () => {
       }));
       const stm = getSearchStateManager();
       stm.initStateFromUrl();
+      // Set list layout since folders layout implies sort to be undefined
+      stm.onLayoutChange(SearchLayout.List);
 
       // Verify that they have been set
       expect(stm.state.query).toBe('hello');
-      expect(stm.state.sort).toBe('alpha-asc');
+      expect(stm.state.sort).toBeUndefined();
```

Comment:
can you explain these changes a bit more? i don't understand. the comment implies that folder layout will make sort `undefined`, but then you set to list layout and sort is still `undefined`. i think if you set list layout **before** calling `initStateFromUrl()` this test can remain the same as it was? 🤔